ambry-tools/src/main/java/com.github.ambry/clustermap/HelixClusterWideAggregationTool.java

```diff
@@ -0,0 +1,94 @@
+ */
+public class HelixClusterWideAggregationTool {
+  private static long WORKFLOW_EXPIRY = TimeUnit.SECONDS.toMillis(30);
+  private static int SESSION_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(30);
+  private static int CONNECTION_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(120);
+  private static String REPORT_NAME = "QuotaReport";
+  private static String TASK_ID = "aggregate_task";
+  private static String JOB_ID = "one_time_aggregate_job";
+
+  /**
+   * @param args takes in three mandatory arguments: the zookeeper address, the cluster name and the workflow name.
+   * @throws Exception
+   */
+  public static void main(String args[]) throws Exception {
+    OptionParser parser = new OptionParser();
+
+    ArgumentAcceptingOptionSpec<String> zkAddressOpt =
+        parser.accepts("zkAddress", "The zookeeper address to connect to")
+            .withRequiredArg()
+            .describedAs("zk_address")
+            .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> clusterNameOpt = parser.accepts("clusterName", "The cluster name in helix")
+        .withRequiredArg()
+        .describedAs("cluster_name")
+        .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> workflowNameOpt =
+        parser.accepts("workflowName", "The name of the one-time workflow")
+            .withRequiredArg()
+            .describedAs("workflow_name")
+            .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<Long> taskTimeoutInSecsOpt =
+        parser.accepts("taskTimeoutInSecs", "Timeout for the task to complete")
+            .withRequiredArg()
+            .describedAs("task_timeout_in_secs")
+            .ofType(Long.class);
+
+    OptionSet options = parser.parse(args);
+    String zkAddress = options.valueOf(zkAddressOpt);
+    String clusterName = options.valueOf(clusterNameOpt);
+    String workflowName = options.valueOf(workflowNameOpt);
+    Long taskTimeoutInSecs = options.valueOf(taskTimeoutInSecsOpt);
+    ZkClient zkClient = new ZkClient(zkAddress, SESSION_TIMEOUT, CONNECTION_TIMEOUT, new ZNRecordSerializer());
+    TaskDriver taskDriver = new TaskDriver(zkClient, clusterName);
+    Workflow.Builder workflowBuilder = new Workflow.Builder(workflowName);
+    workflowBuilder.setExpiry(WORKFLOW_EXPIRY);
+    JobConfig.Builder jobConfigBuilder = new JobConfig.Builder();
+    List<TaskConfig> taskConfigs = new ArrayList<>();
+    taskConfigs.add(new TaskConfig.Builder().setTaskId(TASK_ID)
+        .setCommand(String.format("%s_%s", HelixAggregateTask.TASK_COMMAND_PREFIX, REPORT_NAME))
+        .build());
+    jobConfigBuilder.addTaskConfigs(taskConfigs);
+    workflowBuilder.addJob(JOB_ID, jobConfigBuilder);
+    Workflow workflow = workflowBuilder.build();
+    taskDriver.start(workflow);
+    String jobName = String.format("%s_%s", workflowName, JOB_ID);
+    taskDriver.pollForJobState(workflowName, jobName, taskTimeoutInSecs, TaskState.COMPLETED);
```

Comment:
don't need to wait until the job is complete? This will not return at all. May you can print the jobId info. So that user can check the status of the job in zookeeper.